src/lib/variables/variables.ts

```diff
@@ -25,72 +26,70 @@ export function isClassInstance(input: unknown): boolean {
   );
 }
 
-export function assertIsVariables(
-  input: unknown
-): asserts input is Variables | undefined {
-  if (
-    !['string', 'number', 'boolean', 'object', 'undefined'].includes(
-      typeof input
-    )
-  ) {
-    throw new UnexpectedError(`Invalid result type: ${typeof input}`);
-  }
+export function isNone(input: unknown): input is None {
+  return input === undefined || input === null;
 }
 
-export function castToVariables(input: unknown): Variables | undefined {
-  assertIsVariables(input);
-
-  return input;
-}
-
-export function castToNonPrimitive(input: unknown): NonPrimitive | undefined {
-  const variables = castToVariables(input);
-  if (variables === undefined) {
-    return undefined;
-  }
-
-  if (!isNonPrimitive(variables)) {
-    throw new UnexpectedError('Input is not NonPrimitive');
-  }
-
-  return variables;
-}
-
-export function isPrimitive(input: Variables): input is Primitive {
+export function isPrimitive(input: unknown): input is Primitive {
   return (
     ['string', 'number', 'boolean'].includes(typeof input) ||
     Array.isArray(input) ||
-    isClassInstance(input) ||
+    isNone(input) ||
+    isBinaryData(input) ||
     Buffer.isBuffer(input) ||
-    isBinaryData(input)
+    isClassInstance(input)
   );
 }
 
-export function isNonPrimitive(input: Variables): input is NonPrimitive {
+export function isNonPrimitive(input: unknown): input is NonPrimitive {
   return (
     typeof input === 'object' &&
     !Array.isArray(input) &&
-    !isClassInstance(input) &&
+    !isBinaryData(input) &&
     !Buffer.isBuffer(input) &&
-    !isBinaryData(input)
+    !isClassInstance(input)
```

Comment:
issue: I believe this should also be checking for `input !== null`

Reply:
Indeed, it should. Thank you.